src/main/java/pl/javastart/task/UniversityApp.java

```diff
@@ -42,10 +89,31 @@ public void createGroup(String code, String name, int lecturerId) {
      * @param firstName - imię studenta
      * @param lastName  - nazwisko studenta
      */
-    public void addStudentToGroup(int index, String groupCode, String firstName, String lastName) {
 
+    public void addStudentToGroup(int index, String groupCode, String firstName, String lastName) {
+        Student student =  findStudent(index);
+        Group group = findGroupByCode(groupCode);
+        if (group == null) {
+            System.out.println("Grupa " + groupCode + " nie istnieje");
+        } else if (!group.checkIfStudentExists(index)) {
+            group.addStudent(index, firstName, lastName);
+            if (student == null) {
+                studentsPrintList[studentsPrintListCounter] = new Student(firstName, lastName, index);
+                studentsPrintListCounter++;
+            }
+        } else {
+            System.out.println("Student o indeksie " + index + " jest już w grupie " + groupCode);
+        }
```

Comment:
To można zapisać jeszcze trochę bardziej obiektowo. Zamiast tworzyć obiekt Student osobno w grupie, i osobno tutaj, można by go utworzyć tylko raz
```
Group group = findGroupByCode(groupCode);
if (group == null) {
    System.out.println("Grupa " + groupCode + " nie istnieje");
} else if (!group.checkIfStudentExists(index)) {
    Student student =  findStudent(index);
    if (student == null) {
        student = new Student(firstName, lastName, index);
        studentsPrintList[studentsPrintListCounter] = student;
        studentsPrintListCounter++;
    }
    group.addStudent(index, firstName, lastName);
} else {
    System.out.println("Student o indeksie " + index + " jest już w grupie " + groupCode);
}
```